src/components/__tests__/HeadingForm.test.js

```diff
@@ -0,0 +1,112 @@
+import React from 'react';
+import { fireEvent, cleanup } from 'react-testing-library';
+import 'jest-dom/extend-expect';
+import HeadingForm from '../HeadingForm';
+import { renderWithRedux, testSmData } from '../../services/testing-helpers';
+
+const initialState = {
+  smData: testSmData
+};
+
+afterEach(cleanup);
+
+test('the HeadingForm renders', () => {
+  const { getByTestId } = renderWithRedux(<HeadingForm />);
+  expect(getByTestId('heading-form')).toBeInTheDocument();
+});
+
+test('select child options are available in dropdown', () => {
+  const { container, getByTestId } = renderWithRedux(<HeadingForm />, {
+    initialState
+  });
+  const el = container.querySelector('#headingChildOf');
+  expect(el.children.length).toBe(4);
+  expect(el.children[1].value).toBe('klmn');
+});
+
+test('heading title input shows proper validation messages', async () => {
+  const { container, getByLabelText, getByTestId, debug } = renderWithRedux(
+    <HeadingForm />,
+    {
+      initialState
+    }
+  );
+
+  const titleInput = getByLabelText(/title/i);
```

Comment:
@adamjarling Question: here the `getByLabelText(/title/i)` => is this something like Regex?

Reply:
Yea, case insensitive regex.   They use is all over the place in the docs.